src/components/modal/modal.tsx

```diff
@@ -268,30 +276,41 @@ export class Modal
   }
 
   renderStyle(): VNode {
-    const hasCustomWidth = !isNaN(parseInt(`${this.width}`));
-    return hasCustomWidth ? (
-      <style>
-        {`
-        .${CSS.modal} {
-          max-width: ${this.width}px !important;
-        }
-        @media screen and (max-width: ${this.width}px) {
-          .${CSS.modal} {
-            height: 100% !important;
-            max-height: 100% !important;
-            width: 100% !important;
-            max-width: 100% !important;
-            margin: 0 !important;
-            border-radius: 0 !important;
-          }
-          .content {
-            flex: 1 1 auto !important;
-            max-height: unset !important;
-          }
-        }
-      `}
-      </style>
-    ) : null;
+    if (!this.fullscreen && (this.cssWidth || this.cssHeight)) {
+      return (
+        <style>
```

Comment:
why put this here instead of in the .scss file? better yet, set this with CSS vars rather than static rules.

Reply:
This is using the existing pattern that was setting the `width` (as number) value. The css variable when in unpredictable format (calc, vw, etc) is not compatible with the css file media query, was I believe the original reason for it.